### PR TITLE
Add guardpost for android

### PR DESCRIPTION
**Pull request by rachitmishra**
Adds Guardpost for Android

### Login
- You register an instance of `GuardpostAuthReceiver` with intent filter `GuardpostAuth.BroadcastActions.INTENT_FILTER` using `LocalBroadcastManager`.
  The `GuardpostAuthReceiver` takes callbacks for login/logout/error and invokes them as required. 

- You start the login by calling `GuardpostAuth.startLogin ` by passing the required params.

### Logout
- You start the logout by calling `GuardpostAuth.startLogout `

## Flow
1. `startLogin()` starts a chrome custom tab (CCT).
2. CCT redirects to `RedirectActivity` using deep-links on a successful login/logout call.
3. `RedirectActivity` checks the response and broadcasts success/failure using `LocalBroadcastManager`

## Public API
```
GuardpostAuth.startLogin(context, clientApiKey, nonce) \\ to start login
``` 

 ```
GuardpostAuth.startLogout(context) \\ to get logout intent.
```
```
GuardpostAuth.BroadcastAction.INTENT_FILTER \\ to register for Guardpost broadcast actions.
````
```
GuardpostAuthReceiver \\ Helper class to easily check GuardpostAuth response
````

## Test Scenarios
Tested with logged in user.
Tested with logged out user.

### CODE REVIEW

File: guardpost/src/main/java/com/raywenderlich/guardpost/RedirectActivity.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_redirect)
+
+    val nonce = intent.getStringExtra(EXTRA_NONCE)
+    val clientApiKey = intent.getStringExtra(EXTRA_CLIENT_API_KEY)
+
+    if (intent?.action == ACTION_LOGIN) {
+      guardPostAuth = GuardpostAuth.newInstance(nonce, clientApiKey)
+      guardPostAuth?.startLogin(this)
```
Comment (tgblank) [inline guardpost/src/main/java/com/raywenderlich/guardpost/RedirectActivity.kt]: This appears to duplicate the startLogin call below in the case of ACTION_LOGIN

**Review by tgblank — COMMENTED** [review]
**Issues:**

- If I back out of the login view without doing anything, the activity shows the progress spinner indefinitely.

- We may want to consider using the following customizations on CCT to make it feel more native to the main application:
  - Custom animations with slide in and out
  - Custom toolbar color (possible an option to be set by the main application)
  - Custom close button using back arrow

File: guardpost/src/main/java/com/raywenderlich/guardpost/RedirectActivity.kt (hunk 1 above)
Comment (tgblank) [reply guardpost/src/main/java/com/raywenderlich/guardpost/RedirectActivity.kt]: @rachitmishra Were you able to verify that this is a bug in RedirecTActivity.kt?

Comment (rachitmishra) [reply guardpost/src/main/java/com/raywenderlich/guardpost/RedirectActivity.kt]: yes. I removed this logic too.

**Comment by rachitmishra** [comment]
- [x] Custom animations with slide in and out
- [x] Custom toolbar color (possible an option to be set by the main application)
- [x] Custom close button using back arrow

## Tested

- [x] Login 
- [x] Logout
- [x] Previous login
- [x] Previous login from Chrome

**Review by tgblank — APPROVED** [review]
Nice job @rachitmishra. The changes look good and I like the smooth transition between the native app and the login page.

**Pull request merged**